webapp/src/main/resources/public/js/components/users/UserModal.js

```diff
@@ -0,0 +1,271 @@
+import React from "react";
+import {FormattedMessage, injectIntl} from "react-intl";
+import {Button, FormControl, Modal, Form, FormGroup, ControlLabel, Alert, Collapse, Glyphicon} from "react-bootstrap";
+import UserStatics from "../../utils/UserStatics";
+import UserActions from "../../actions/users/UserActions";
```

Comment:
Big thanks for doing the refactoring for using "alt"! 

One last suggestion, in other pages, the actions are also passed through props eg.

```
 <AltContainer store={ScreenshotsLocaleStore}>
                                <LocalesDropdown
                                    onSelectedBcp47TagsChanged={(selectedBcp47Tags) => {
                                        ScreenshotsHistoryActions.disableHistoryUpdate();
                                        ScreenshotsPaginatorActions.changeCurrentPageNumber(1);
                                        ScreenshotsLocaleActions.changeSelectedBcp47Tags(selectedBcp47Tags);
                                        ScreenshotsHistoryActions.enableHistoryUpdate();
                                        ScreenshotsPageActions.performSearch();
                                        }}
                                    onDropdownToggle={ScreenshotsLocaleActions.changeDropdownOpen}/>
                            </AltContainer>
```

This way the UI component is truly decoupled from the store (that should help when we want to move say to redux) and can potentially be reused by different page

Reply:
> This way the UI component is truly decoupled from the store

Only theoretically. In practice you likely won't be able to swap them out and will only use the `UserStore`.

> and can potentially be reused by different page

I highly doubt this will be possible, since the components are pretty specific.

> that should help when we want to move say to redux

From what I can tell from https://react-redux.js.org/tutorials/quick-start, this shouldn't be required. As far as I understand the tutorial, you use the slice directly in the component without the props.

Reply:
agree for the user components it makes little sense. 

That's applicable for components like the LocalesDropdown which should be the same for all pages but if actions are hardcoded in it then it is problematic. Then I guess it boils down to wanting to apply the same pattern everywhere. 

This is mostly suggestions at this points, i'm happy to merge as it if you think that's the way to go

Reply:
I have changed it for the `UserErrorModal` and the `UserDeleteModal`, since it does make sense here.

However, I would be against also changing it for the other components. Especially in the `UserEditModal`, where I would have to use a prop for updating every input field. Moving these calls to props would make the code harder to understand IMHO.